manifests/server/resources/bootstrap.pp

```diff
@@ -1,13 +1,26 @@
 # @summary Generates and deploys the default Puppet boostrap configuration into the cluster
 #
 # @param master The main API URL to encode in the bootstrap configuration
+# @param secret The exact token secret to use, will be generated as a random 16-char string if left blank.
+#   The generated value can be retrieved from the bootstrap-token-puppet Secret in kube-system.
 class k8s::server::resources::bootstrap (
   K8s::Ensure $ensure                    = $k8s::ensure,
   Stdlib::Unixpath $kubeconfig           = $k8s::server::resources::kubeconfig,
   String[1] $master                      = $k8s::server::resources::master,
+
+  Optional[K8s::Bootstrap_token] $secret = undef,
```

Comment:
would it be possible or does it make sense to use the Sensitive-type here?

Reply:
The K8s::Bootstrap_token type could potentially be a Sensitive, yes. I'll need to test and see what difference it does to the generation and storage as part of the kubectl_apply though.